### ISSUE TITLE
Multiremote test crashes with allure reporter enabled

### ISSUE THREAD

**Issue by slysenko** [webdriverio/wdio-allure-reporter#117]
>> Description:

wdio/jasmine test with multiremote config and test suite based on http://webdriver.io/guide/usage/multiremote.html

```
var mainConfig = require('./wdio.conf');

mainConfig.config.capabilities = {
  macFF: {
    desiredCapabilities: {
      browserName: 'firefox',
      platform: "MAC"
    }
  },
   winFF: {
     desiredCapabilities: {
       browserName: 'firefox',
       platform: "WINDOWS"
     }
   }
};

mainConfig.config.reporters = ['allure', 'junit', 'spec'];

module.exports = mainConfig;
```

in package.json: 
```"wdio-allure-reporter": "^0.6.0",```

with 'allure' included in 'reporters, the test crashes with the following errors:

```
./node_modules/.bin/wdio ./wdio.conf.test.inheritance.multiremote.js --spec ./specs/test.multiremote.js --host localhost
let's go
/Users/test/Projects/web/Sources/MyFrontend/autotest/node_modules/webdriverio/build/lib/utils/BaseReporter.js:336
                        throw _iteratorError;
                        ^

TypeError: Cannot read property 'uri' of undefined
    at AllureReporter.runnerResult (/Users/test/Projects/web/Sources/MyFrontend/autotest/node_modules/wdio-allure-reporter/build/reporter.js:254:40)
    at emitOne (events.js:116:13)
    at AllureReporter.emit (events.js:211:7)
    at BaseReporter.handleEvent (/Users/test/Projects/web/Sources/MyFrontend/autotest/node_modules/webdriverio/build/lib/utils/BaseReporter.js:324:35)
    at Launcher.messageHandler (/Users/test/Projects/web/Sources/MyFrontend/autotest/node_modules/webdriverio/build/lib/launcher.js:624:28)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at emit (internal/child_process.js:772:12)
    at _combinedTickCallback (internal/process/next_tick.js:141:11)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9)
/Users/test/Projects/web/Sources/MyFrontend/autotest/node_modules/q/q.js:155
                throw e;
                ^

Error [ERR_IPC_CHANNEL_CLOSED]: channel closed
    at process.target.send (internal/child_process.js:588:16)
    at EventEmitter.<anonymous> (/Users/test/Projects/web/Sources/MyFrontend/autotest/node_modules/webdriverio/build/lib/runner.js:171:45)
    at emitOne (events.js:121:20)
    at EventEmitter.emit (events.js:211:7)
    at Object.prototype.(anonymous function) [as emit] (/Users/test/Projects/web/Sources/MyFrontend/autotest/node_modules/webdriverio/build/lib/webdriverio.js:783:40)
    at /Users/test/Projects/web/Sources/MyFrontend/autotest/node_modules/webdriverio/build/lib/multibrowser.js:163:23
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
```

if 'allure' is removed from the 'reporters' the test goes as expected, i.e

```
let's go
{ macFF: 'Google', winFF: 'Google' }
Yahoo is now a part of Oath
{ macFF: 'Google', winFF: 'Yahoo is now a part of Oath' }
------------------------------------------------------------------
[undefined #0-0] Session ID: undefined
[undefined #0-0] Spec: /Users/slysenko/Projects/WORK/web/Sources/MyAccount/Frontend/my/autotest/specs/test.multiremote.js
[undefined #0-0] Running: undefined
[undefined #0-0]
[undefined #0-0] test
[undefined #0-0]   ✓ should do something with two browser
[undefined #0-0]
[undefined #0-0]
[undefined #0-0] 1 passing (22s)
[undefined #0-0]

Wrote xunit report to [./junit-reports].
1 passing (22.40s)
```

**Comment by BorisOsipov** [comment]
@slysenko can you [enable verbose log](http://webdriver.io/guide/getstarted/configuration.html#logLevel) and share it with us?

**Comment by BorisOsipov** [comment]
@slysenko or share example of project.

**Comment by slysenko** [comment]
```
10:36:46.051 INFO - Found handler: org.openqa.selenium.remote.server.commandhandler.BeginSession@2911c5bd
10:36:46.051 INFO - /session: Executing POST on /session (handler: BeginSession)
10:36:46.052 INFO - Found handler: org.openqa.selenium.remote.server.commandhandler.BeginSession@aaf7f11
10:36:46.052 INFO - /session: Executing POST on /session (handler: BeginSession)
10:36:46.055 INFO - Capabilities are: Capabilities {browserName: firefox, handlesAlerts: true, javascriptEnabled: true, locationContextEnabled: true, loggingPrefs: org.openqa.selenium.logging..., platform: MAC, requestOrigins: {name: webdriverio, url: http://webdriver.io, version: 4.9.8}, rotatable: true}
10:36:46.055 INFO - Capabilities are: Capabilities {browserName: chrome, handlesAlerts: true, javascriptEnabled: true, locationContextEnabled: true, loggingPrefs: org.openqa.selenium.logging..., platform: MAC, requestOrigins: {name: webdriverio, url: http://webdriver.io, version: 4.9.8}, rotatable: true}
10:36:46.055 INFO - Capabilities {browserName: firefox, handlesAlerts: true, javascriptEnabled: true, locationContextEnabled: true, loggingPrefs: org.openqa.selenium.logging..., platform: MAC, requestOrigins: {name: webdriverio, url: http://webdriver.io, version: 4.9.8}, rotatable: true} matched class org.openqa.selenium.remote.server.ServicedSession$Factory (provider: org.openqa.selenium.firefox.GeckoDriverService)
10:36:46.055 INFO - Capabilities {browserName: chrome, handlesAlerts: true, javascriptEnabled: true, locationContextEnabled: true, loggingPrefs: org.openqa.selenium.logging..., platform: MAC, requestOrigins: {name: webdriverio, url: http://webdriver.io, version: 4.9.8}, rotatable: true} matched class org.openqa.selenium.remote.server.ServicedSession$Factory (provider: org.openqa.selenium.chrome.ChromeDriverService)
1527320206072   geckodriver     INFO    geckodriver 0.20.0
1527320206079   geckodriver     INFO    Listening on 127.0.0.1:36971
Starting ChromeDriver 2.37.544337 (8c0344a12e552148c185f7d5117db1f28d6c9e85) on port 28673
Only local connections are allowed.
1527320206198   mozrunner::runner       INFO    Running command: "/Applications/Firefox.app/Contents/MacOS/firefox-bin" "-marionette" "-profile" "/var/folders/yr/b4_dyc6s0l74d73fbrqk9h9w0000gp/T/rust_mozprofile.9GNoZaBbLHUI"
1527320207068   Marionette      INFO    Enabled via --marionette
10:36:47.312 INFO - Detected dialect: OSS
2018-05-26 10:36:48.077 plugin-container[91163:51538759] *** CFMessagePort: bootstrap_register(): failed 1100 (0x44c) 'Permission denied', port = 0xb437, name = 'com.apple.tsm.portname'
See /usr/include/servers/bootstrap_defs.h for the error codes.
1527320208468   Marionette      INFO    Listening on port 49990
10:36:48.591 INFO - Detected dialect: W3C
10:36:48.634 INFO - Found handler: org.openqa.selenium.remote.server.ServicedSession@7960a485
10:36:48.635 INFO - Handler thread for session 3c7165ca-c3fb-f941-b8fc-09be5c6e8de7 (firefox): Executing POST on /session/3c7165ca-c3fb-f941-b8fc-09be5c6e8de7/url (handler: ServicedSession)
10:36:48.635 INFO - Found handler: org.openqa.selenium.remote.server.ServicedSession@2311c54
10:36:48.635 INFO - Handler thread for session bc3f8ad9211c20c023b8f2814a8b1ac8 (chrome): Executing POST on /session/bc3f8ad9211c20c023b8f2814a8b1ac8/url (handler: ServicedSession)
10:36:48.635 INFO - To upstream: {"url":"http://google.com/"}
2018-05-26 10:36:48.940 plugin-container[91165:51538891] *** CFMessagePort: bootstrap_register(): failed 1100 (0x44c) 'Permission denied', port = 0xae37, name = 'com.apple.tsm.portname'
See /usr/include/servers/bootstrap_defs.h for the error codes.
10:36:49.871 INFO - To downstream: {"sessionId":"bc3f8ad9211c20c023b8f2814a8b1ac8","status":0,"value":null}
10:36:50.932 INFO - Found handler: org.openqa.selenium.remote.server.ServicedSession@7960a485
10:36:50.932 INFO - Found handler: org.openqa.selenium.remote.server.ServicedSession@2311c54
10:36:50.932 INFO - Handler thread for session bc3f8ad9211c20c023b8f2814a8b1ac8 (chrome): Executing GET on /session/bc3f8ad9211c20c023b8f2814a8b1ac8/title (handler: ServicedSession)
10:36:50.932 INFO - Handler thread for session 3c7165ca-c3fb-f941-b8fc-09be5c6e8de7 (firefox): Executing GET on /session/3c7165ca-c3fb-f941-b8fc-09be5c6e8de7/title (handler: ServicedSession)
10:36:50.937 INFO - To downstream: {"sessionId":"bc3f8ad9211c20c023b8f2814a8b1ac8","status":0,"value":"Google"}
10:36:51.050 INFO - Found handler: org.openqa.selenium.remote.server.ServicedSession@2311c54
10:36:51.050 INFO - Found handler: org.openqa.selenium.remote.server.ServicedSession@7960a485
10:36:51.050 INFO - Handler thread for session bc3f8ad9211c20c023b8f2814a8b1ac8 (chrome): Executing DELETE on /session/bc3f8ad9211c20c023b8f2814a8b1ac8 (handler: ServicedSession)
10:36:51.050 INFO - Handler thread for session 3c7165ca-c3fb-f941-b8fc-09be5c6e8de7 (firefox): Executing DELETE on /session/3c7165ca-c3fb-f941-b8fc-09be5c6e8de7 (handler: ServicedSession)
1527320211068   addons.xpi      WARN    Exception running bootstrap method shutdown on activity-stream@mozilla.org: [Exception... "Component returned failure code: 0x80004005 (NS_ERROR_FAILURE) [nsIObserverService.removeObserver]"  nsresult: "0x80004005 (NS_ERROR_FAILURE)"  location: "JS frame :: resource://activity-stream/lib/SnippetsFeed.jsm :: uninit :: line 125"  data: no]Stack trace: uninit()@resource://activity-stream/lib/SnippetsFeed.jsm:125 < onAction()@resource://activity-stream/lib/SnippetsFeed.jsm:141 < _middleware/</<()@resource://activity-stream/lib/Store.jsm:51 < Store/this[method]()@resource://activity-stream/lib/Store.jsm:30 < uninit()@resource://activity-stream/lib/Store.jsm:153 < uninit()@resource://activity-stream/lib/ActivityStream.jsm:274 < uninit()@resource://gre/modules/addons/XPIProvider.jsm -> jar:file:///Applications/Firefox.app/Contents/Resources/browser/features/activity-stream@mozilla.org.xpi!/bootstrap.js:80 < shutdown()@resource://gre/modules/addons/XPIProvider.jsm -> jar:file:///Applications/Firefox.app/Contents/Resources/browser/features/activity-stream@mozilla.org.xpi!/bootstrap.js:196 <callBootstrapMethod()@resource://gre/modules/addons/XPIProvider.jsm:4419 < observe()@resource://gre/modules/addons/XPIProvider.jsm:2279 < GeckoDriver.prototype.quit()@driver.js:3270 < despatch()@server.js:557 < execute()@server.js:531 < onPacket/<()@server.js:506 < onPacket()@server.js:505 < _onJSONObjectReady/<()@transport.js:500
10:36:51.102 INFO - To downstream: {"sessionId":"bc3f8ad9211c20c023b8f2814a8b1ac8","status":0,"value":null}
10:36:51.102 INFO - Removing session org.openqa.selenium.remote.server.ServicedSession@2311c54
10:36:51.112 INFO - To downstream: {"sessionId":"","status":0,"value":null}
*** UTM:SVC TimerManager:registerTimer called after profile-before-change notification. Ignoring timer registration for id: telemetry_modules_ping
1527320211635   geckodriver::marionette ERROR   Failed to stop browser process
10:36:51.645 INFO - Removing session org.openqa.selenium.remote.server.ServicedSession@7960a485
```

**Comment by slysenko** [comment]
this is from selenium-standalone

**Comment by slysenko** [comment]
```
describe('test', function() {

  it('should do something with two browser', function() {
    browser.url('http://google.com');
    console.log(browser.getTitle());

    macCh.url('http://yahoo.com');
    console.log(macCh.getTitle());

    console.log(browser.getTitle());
  });
});
```

```
var mainConfig = require('./wdio.conf');

mainConfig.config.capabilities = {
  macFF: {
    desiredCapabilities: {
      browserName: 'firefox',
      platform: "MAC"
    }
  },
  macCh: {
    desiredCapabilities: {
      browserName: 'chrome',
      platform: "MAC"
    }
  },
};

mainConfig.config.reporters = ['allure', 'junit', 'spec'];

module.exports = mainConfig;
```

**Comment by slysenko** [comment]
```
[10:42:36]  COMMAND	POST 	 "/wd/hub/session"
[10:42:36]  DATA		{"desiredCapabilities":{"javascriptEnabled":true,"locationContextEnabled":true,"handlesAlerts":true,"rotatable":true,"browserName":"chrome","platform":"MAC","loggingPrefs":{"browser":"ALL","driver":"ALL"},"requestOrigins":{"url":"http://webdriver.io","version":"4.9.8","name":"webdriverio"}}}
[10:42:38]  INFO	SET SESSION ID 1c35072458f26e7ff82f6b2228d0bdb8
[10:42:38]  RESULT		{"applicationCacheEnabled":false,"rotatable":false,"mobileEmulationEnabled":false,"networkConnectionEnabled":false,"chrome":{"chromedriverVersion":"2.38.552518 (183d19265345f54ce39cbb94cf81ba5f15905011)","userDataDir":"/var/folders/hn/22l6gt754f5_ss0dq327rccc0000gn/T/.org.chromium.Chromium.2bkSPa"},"takesHeapSnapshot":true,"pageLoadStrategy":"normal","databaseEnabled":false,"handlesAlerts":true,"hasTouchScreen":false,"version":"66.0.3359.181","platform":"Mac OS X","browserConnectionEnabled":false,"nativeEvents":true,"acceptSslCerts":false,"acceptInsecureCerts":false,"locationContextEnabled":true,"webStorageEnabled":true,"browserName":"chrome","takesScreenshot":true,"javascriptEnabled":true,"cssSelectorsEnabled":true,"setWindowRect":true,"unexpectedAlertBehaviour":""}
[10:42:41]  COMMAND	POST 	 "/wd/hub/session/1c35072458f26e7ff82f6b2228d0bdb8/url"
[10:42:41]  DATA		{"url":"http://google.com/"}
[10:42:44]  COMMAND	GET 	 "/wd/hub/session/1c35072458f26e7ff82f6b2228d0bdb8/title"
[10:42:44]  DATA		{}
[10:42:44]  RESULT		"Google"
[10:42:44]  COMMAND	DELETE 	 "/wd/hub/session/1c35072458f26e7ff82f6b2228d0bdb8"
[10:42:44]  DATA		{}
```

**Comment by slysenko** [comment]
```
[10:42:36]  COMMAND	POST 	 "/wd/hub/session"
[10:42:36]  DATA		{"desiredCapabilities":{"javascriptEnabled":true,"locationContextEnabled":true,"handlesAlerts":true,"rotatable":true,"browserName":"firefox","platform":"MAC","loggingPrefs":{"browser":"ALL","driver":"ALL"},"requestOrigins":{"url":"http://webdriver.io","version":"4.9.8","name":"webdriverio"}}}
[10:42:41]  INFO	SET SESSION ID f4cbb740-092b-454d-b68c-e4e0e96c4ade
[10:42:41]  RESULT		{"sessionId":"f4cbb740-092b-454d-b68c-e4e0e96c4ade","capabilities":{"moz:profile":"/var/folders/hn/22l6gt754f5_ss0dq327rccc0000gn/T/rust_mozprofile.KMpxlEj3L56z","rotatable":false,"timeouts":{"implicit":0,"pageLoad":300000,"script":30000},"pageLoadStrategy":"normal","moz:headless":false,"moz:accessibilityChecks":false,"moz:useNonSpecCompliantPointerOrigin":false,"acceptInsecureCerts":false,"browserVersion":"59.0.2","platformVersion":"17.5.0","moz:processID":575,"browserName":"firefox","platformName":"darwin","moz:webdriverClick":true}}
[10:42:41]  COMMAND	POST 	 "/wd/hub/session/f4cbb740-092b-454d-b68c-e4e0e96c4ade/url"
[10:42:41]  DATA		{"url":"http://google.com/"}
[10:42:44]  RESULT		{}
[10:42:44]  COMMAND	GET 	 "/wd/hub/session/f4cbb740-092b-454d-b68c-e4e0e96c4ade/title"
[10:42:44]  DATA		{}
[10:42:44]  RESULT		"Google"
[10:42:44]  COMMAND	DELETE 	 "/wd/hub/session/f4cbb740-092b-454d-b68c-e4e0e96c4ade"
[10:42:44]  DATA		{}
[10:42:45]  RESULT		{}
```

**Comment by slysenko** [comment]
package.json

```
  "dependencies": {
    "chai": "^3.5.0",
    "htmlparser": "^1.7.7",
    "imap": "^0.8.19",
    "jasmine": "^2.9.0",
    "minimist": "^1.2.0",
    "soupselect": "^0.2.0",
    "ssh2": "^0.5.5",
    "wdio-allure-reporter": "^0.6.0",
    "wdio-jasmine-framework": "0.3.2",
    "wdio-junit-reporter": "^0.3.1",
    "wdio-selenium-standalone-service": "^0.0.9",
    "wdio-spec-reporter": "0.1.2",
    "webdriverio": "4.9.8"
  },
```

**Comment by christian-bromann** [comment]
Fixed by #118
